### ISSUE TITLE
Kernel v0.1.10 resets my account balance after a while

### ISSUE THREAD

**Issue by JimVanEeden** [aionnetwork/aion#16]
This happened twice now:

```
aion-127.0.0.1:8545> eth.getBalance('0xf7f1aa76f73c1d6a7d09c19c002b3e2acc8a81fe96d68afed0a37111a806c313')
BigNumber { s: 1, e: 16, c: [ 152, 8333333333236 ] }
aion-127.0.0.1:8545> eth.getBalance('0xf7f1aa76f73c1d6a7d09c19c002b3e2acc8a81fe96d68afed0a37111a806c313')
BigNumber { s: 1, e: 16, c: [ 226, 62037037036892 ] }
aion-127.0.0.1:8545> eth.getBalance('0xf7f1aa76f73c1d6a7d09c19c002b3e2acc8a81fe96d68afed0a37111a806c313')
0
aion-127.0.0.1:8545> eth.getBalance('0xc76dd015be0ba336821e3eb594a157c34cdae74e8ba65e2f76cb33b10d26a7fa')
BigNumber { s: 1, e: 16, c: [ 232, 92824074073924 ] }
aion-127.0.0.1:8545> eth.getBalance('0xc76dd015be0ba336821e3eb594a157c34cdae74e8ba65e2f76cb33b10d26a7fa')
BigNumber { s: 1, e: 16, c: [ 249, 99999999999840 ] }
aion-127.0.0.1:8545> eth.getBalance('0xc76dd015be0ba336821e3eb594a157c34cdae74e8ba65e2f76cb33b10d26a7fa')
BigNumber { s: 1, e: 16, c: [ 267, 12962962962790 ] }
aion-127.0.0.1:8545> eth.getBalance('0xc76dd015be0ba336821e3eb594a157c34cdae74e8ba65e2f76cb33b10d26a7fa')
0
```

The first account I just added to the config/config.xml in the mining section. After I noticed it went to 0 I created a new account through the kernel and then replaced the first account with this new one in config/config.xml. Again the account went to 0 after a while.

I'm not seeing any error messages appear in the kernel miner, but it is possible there were some because I can't scroll that far up anymore.

**Comment by AlexandraRoatis** [comment]
After some thorough investigation, I suspect the loss of balance is because the blocks that you are mining are not always included in the main chain. The mining reward is given only to the blocks on the main chain. So if you mine a block you will temporarily be given the coins for it, but if the consensus algorithm later decides that someone else mined that block first, then the balance is adjusted and only the miner whose block was included in the main chain gets the reward.

To keep better track of this phenomenon, you can edit the `config.xml` file to set:
```
<p2p>
    [...]
     <show-status>true</show-status>
</p2p>
```
This will allow you to see the status of your peer connections.
If you see:
```
[active-nodes(nodeIdHash)=[]]
```
You've lost connection to peers, and very likely all the blocks you are mining will be on a side chain, hence the rewards will be lost when reconnecting to the network.

Further, provided that your config file has the following settings:
```
<SYNC>INFO</SYNC>
<CONS>INFO</CONS>
```
you can observe when your block has **not been included** in the main chain when you see messages like the following:
```
INFO  CONS [ConsHdr]: block sealed <num=100, hash=1e9ce9, diff=62, tx=0>
INFO  CONS [sync-import]: branching: <num=100 prev=1e9ce9 -> to=575f44>
INFO  SYNC [sync-import]: <import-best num=100 hash=575f44 txs=0>
```
The first line marks that you have created a new block. The next two lines show that your block has not been used by the main chain. A block with the same number but different hash (hence mined by someone else) has been chosen.

**Note:** Other networks also reward side chains, but this test net does not.

**Comment by JimVanEeden** [comment]
Ok so I do think that you are right about what you are saying. Starting the kernel (2 days ago) resulted in it start mining straight away from block 1, which I found odd. So not being able to connect to the peers for some reason might explain this and then why the balance went to 0 after a while.

Sadly I can't reproduce this now, because when I started it this morning it started syncing like normally, getting up to block 141303. With the logs showing the following:

```
[p2p-status iXDEV]
[temp-nodes-size=26278]
[inbound-nodes-size=0]
[outbound-nodes-size=0]
[active-nodes(nodeIdHash)=[                   <---- good?
	144351-0466a-191.232.176.213,
	152945-2da62-52.237.31.69,
	150372-d9242-13.95.218.95]]

[p2p-status iXDEV]
[temp-nodes-size=26277]
[inbound-nodes-size=0]
[outbound-nodes-size=0]
[active-nodes(nodeIdHash)=[                 <---- not good?
	144351-0466a-191.232.176.213,
	150372-d9242-13.95.218.95]]
```

However, the kernel now doesn't start mining anymore. (This might be a separate issue.) The only thing I can see is that at some point after the syncing it loses connection to one peer and it never syncs anymore or starts mining. Only when I restart the kernel it syncs again, but hangs again after it's done syncing. I can make a separate issue on this if needed.

**Comment by AlexandraRoatis** [comment]
You're correct, losing connection to the main chain is a problem. However, we've made significant progress in solving the sync issues and a new binary will be available by the end of the week.

For the time being, if you lose track of peers, you should close and restart the Aion instance. To ensure that mining is enabled please check the `config.xml` file and ensure that the mining tag is set to true.
```
<consensus>
	<mining>true</mining>
        [...]
</consensus>
```
In general, actually mining a block will not occur very frequently because other computers might get the solution faster, so the instance gets the message that the block has been mined already and aims for mining the next block.

**Note:** You didn't experience these issues with the old kernel mainly because the number of peers on the network was smaller at the time. When using the next binary you can fast-track the sync process by copying the full database folder from the old instance into the new one.

**Comment by JimVanEeden** [comment]
My mining was enabled all the time. However, I finally got it working again. Had to restart it a couple of times until I saw this:

```
[active-nodes(nodeIdHash)=[
	157324-3856c-52.161.92.4,
	157968-c1f42-52.179.100.107,
	157966-0466a-191.232.176.213,
	157965-2da62-52.237.31.69,
	157969-d9242-13.95.218.95]]
```
With these connections it actually started mining. For some reason it doesn't always connect to all nodes apparently. But restarting a few times helps. Thanks for your help  and I hope the new release fixes the issue :)
